### ISSUE TITLE
divisionBy scale issue!

### ISSUE THREAD

**Issue by mmuruev** [brick/math#6]
I've found interesting thing about it
http://screencloud.net/v/7zDR
```
(string) $rate->plus(1)->power($nper)->minus(1)->dividedBy('0.010', null, Brick\Math\RoundingMode::CEILING); // works correctly!
(string) $rate->plus(1)->power($nper)->minus(1)->dividedBy('0.01', null, Brick\Math\RoundingMode::CEILING) ; // incorrect comma should be shifted but nothing happened! 
```

**Comment by BenMorel** [comment]
Indeed, this was a gross bug in an optimization. Thanks so much for reporting it!
It's fixed now, and I just released version [0.5.3](https://github.com/brick/math/releases/tag/0.5.3) that includes the fix.

**Comment by BenMorel** [comment]
This was a pretty big issue, so the fix is backported to `0.3` and `0.4` as well.
Previous versions are not affected as they did not include optimizations.
